### ISSUE TITLE
Improve Accessibility to meet WCAG 2.1 AA

### ISSUE THREAD

**Issue by rcheetham** [opendataphilly/opendataphilly-jkan#592]
Based on an initial pass from Google Lighthouse (see #578) we got a 91 score on mobile and 88 score on desktop. It highlighted the following items:

- `<html>` element does not have a `[lang]` attribute - more at: https://dequeuniversity.com/rules/axe/4.11/html-has-lang - seems easy to fix by adding `lang="en"'`
- Heading elements are not in a sequentially-descending order - our "Browse by Category" header is H3 after H1 (skipping H2). On our interior pages, we just jump to H3 on a lot of headings with no intervening H1 or H2 - more at https://dequeuniversity.com/rules/axe/4.11/heading-order
- Image elements do not have `[alt]` attributes that are redundant text. We have a few failing images with redundant text (the `ALT` has identical text to the adjacent text, and this confuses screen readers
<img width="906" height="506" alt="Image" src="https://github.com/user-attachments/assets/2beda31c-8e60-4138-83a4-e3b0f4d931fa" />
- Background and foreground colors do not have a sufficient contrast ratio - this is flagged on interior pages (like /datasets/) but not on the home page - while our dark gray is high contrast, the teal we use for our links on hover is only has a 2.15:1 contrast ratio and it's 3.9:1 for the default (not hover) teal. We need to have a 4.5:1 to pass WCAG 2.1 AA - we could potentially keep the same teal color palette and make the color darker and perhaps not make as big of a change on hover. 
- Document does not have a main landmark - we need to have clear `<header>`, `<nav>`, `<main>`, `<footer>` - more at https://dequeuniversity.com/rules/axe/4.11/landmark-one-main - Here is the Deque ideal.:
<img width="861" height="517" alt="Image" src="https://github.com/user-attachments/assets/c56c2180-ca22-4aee-a0a5-428365ef2483" />

There are also a dozen or so things that have to be checked manually. 

Make the following changes:
- [ ] Add `lang="en"` to `<html>` tags
- [ ] Remove redundant `ALT` tags on category images - these appear on the home page as well as category pages
- [ ] Change use of `H1`, `H2`, `H3` tags so that they are sequential on a given page (if used) or replace `H3` usage with a CSS style that doesn't imply hierarchical structure.
- [ ] Adjust colors for links: If we use the same color scheme , then change default color to `367D87` and hover color to `168383`. Alternatively, we could remove the color change on hover.
- [ ] Adjust landmark HTML tags on pages: `<header>`, `<nav>`, `<main>`, `<footer>` - I think the report is flagging that our landmark HTML tags are a bit jumbled or missing. Our `<footer>` tag is inside the `<body>`, rather than after it.  Our `<nav>` tag is inside the `<body>`, rather than before it. 
- [ ] Manually check the following:
   - [x] Interactive controls are keyboard focusable
   - [x] Interactive elements indicate their purpose and state
   - [x] The page has a logical tab order
   - [x] Visual order on the page follows DOM order
   - [x] User focus is not accidentally trapped in a region
   - [ ] The user's focus is directed to new content added to the page
   - [x] HTML5 landmark elements are used to improve navigation
   - [ ] Offscreen content is hidden from assistive technology
   - [ ] Custom controls have associated labels
   - [ ] Custom controls have ARIA roles

**Comment by rcheetham** [comment]
Re-ran [Google Lighthouse](https://developer.chrome.com/docs/lighthouse) and we've improved our scores such that:
- Desktop: score of 88 went to 97
- Mobile: score of 91 went to 98

The last item is adding another element tag `<main>`

**Comment by rcheetham** [comment]
The latest PR pushed us to a score of 100 on both Desktop and Mobile.

Results: https://pagespeed.web.dev/analysis/https-opendataphilly-org/dhg93gi3ln
